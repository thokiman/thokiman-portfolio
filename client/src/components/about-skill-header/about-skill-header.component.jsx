import React from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import {
  selectSkillEngineerRoute,
  selectSkillRoute,
  selectSkillTechnologyRoute,
  selectSkillVisualArtRoute,
} from "../../redux/about/about.selectors";
import {
  AboutHeaderSkillContainer,
  AboutTextBox,
} from "./about-skill-header.styles";

export const SkillHeader = ({
  location: { pathname },
  skillRoute,
  visualArtRoute,
  engineerRoute,
  technologyRoute,
}) => {
  return (
    <AboutHeaderSkillContainer>
      <AboutTextBox
        to={`${visualArtRoute}`}
        $matchpath={!!pathname.match(`(${skillRoute}$|${visualArtRoute}$)`)}
      >
        Visual Art
      </AboutTextBox>
      <AboutTextBox
        to={`${engineerRoute}`}
        $matchpath={!!pathname.match(`${engineerRoute}$`)}
      >
        Engineer
      </AboutTextBox>
      <AboutTextBox
        to={`${technologyRoute}`}
        $matchpath={pathname.match(`${technologyRoute}`)}
      >
        Technology
      </AboutTextBox>
    </AboutHeaderSkillContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  skillRoute: selectSkillRoute,
  visualArtRoute: selectSkillVisualArtRoute,
  engineerRoute: selectSkillEngineerRoute,
  technologyRoute: selectSkillTechnologyRoute,
});

export default withRouter(connect(mapStateToProps)(SkillHeader));
