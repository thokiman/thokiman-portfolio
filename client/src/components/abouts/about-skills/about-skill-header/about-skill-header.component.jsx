import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import {
  selectSkillEngineerRoute,
  selectSkillRoute,
  selectSkillTechnologyRoute,
  selectSkillVisualArtRoute,
} from '../../../../redux/about/about.selectors';
import {
  AboutHeaderSkillContainer,
  AboutTextBox,
} from './about-skill-header.styles';

export const SkillHeader = ({
  location: { pathname },
  skillRoute,
  visualArtRoute,
  engineerRoute,
  technologyRoute,
}) => {
  return (
    <AboutHeaderSkillContainer>
      <AboutTextBox
        className='about-text-box'
        to={`${visualArtRoute}`}
        $matchpath={!!pathname.match(`(${skillRoute}$|${visualArtRoute}$)`)}
      >
        Visual Art
      </AboutTextBox>
      <AboutTextBox
        className='about-text-box'
        to={`${engineerRoute}`}
        $matchpath={!!pathname.match(`${engineerRoute}$`)}
      >
        Engineer
      </AboutTextBox>
      <AboutTextBox
        className='about-text-box'
        to={`${technologyRoute}`}
        $matchpath={pathname.match(`${technologyRoute}`)}
      >
        Technology
      </AboutTextBox>
    </AboutHeaderSkillContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  skillRoute: selectSkillRoute,
  visualArtRoute: selectSkillVisualArtRoute,
  engineerRoute: selectSkillEngineerRoute,
  technologyRoute: selectSkillTechnologyRoute,
});

export default withRouter(connect(mapStateToProps)(SkillHeader));
