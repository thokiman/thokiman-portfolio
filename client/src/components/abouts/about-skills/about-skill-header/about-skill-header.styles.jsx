import { Link } from 'react-router-dom';
import styled, { css, keyframes } from 'styled-components';

const lightBlueColor = '#74c6d4';
const maxWidthSmartphone = 800;

export const AboutHeaderSkillContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  height: 10%;
  width: 60vw;
  padding: 15px;
  margin-bottom: 10px;

  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 100%;
    height: 40px;
  }
`;

const AboutTextBoxStyles = css`
  text-decoration: none;
  padding: 5px;
  margin: 0 5px;
  color: ${lightBlueColor};
  font-family: futura-book;
  font-size: 20px;
  letter-spacing: 0.03em;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 20px;
  }
`;
const AboutTextBoxActive = css`
  ${AboutTextBoxStyles}
  cursor: default;
  transition: all 0.7s linear;
  box-shadow: ${lightBlueColor} 0px 20px 21px -9px;
`;
const aboutTextBoxHover = keyframes`
    from{
      opacity:0
    }
    to{
      opacity:1
    }
`;
const AboutTextBoxHide = css`
  ${AboutTextBoxStyles}
  transition: all 0.7s linear;
   &:hover {
    
           transition: all 0.7s linear;
            box-shadow: ${lightBlueColor} 0px 20px 21px -9px;
            animation: ${aboutTextBoxHover} 0.5s ease-in-out both;
            
`;

const getAboutTextBox = ({ $matchpath }) => {
  if ($matchpath) {
    return AboutTextBoxActive;
  }
  return AboutTextBoxHide;
};

export const AboutTextBox = styled(Link)`
  ${getAboutTextBox}
`;

AboutHeaderSkillContainer.displayName = 'AboutHeaderSkillContainer';
AboutTextBox.displayName = 'AboutTextBox';
