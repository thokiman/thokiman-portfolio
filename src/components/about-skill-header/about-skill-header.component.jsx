import React from "react";
import { Link, withRouter } from "react-router-dom";

import "./about-skill-header.styles.scss";

const SkillHeader = ({ match: { url }, location: { pathname } }) => {
  return (
    <div className="header-skill-container">
      <div className="skill-box">
        <Link
          to={`${url}/visualart`}
          className={`${
            pathname.match(/\/about\/skill\/visualart/)
              ? "active-text-box"
              : "text-box"
          }`}
        >
          Visual Art
        </Link>
      </div>
      <div className="skill-box">
        <Link
          to={`${url}/engineer`}
          className={`${
            pathname.match(/\/about\/skill\/engineer/)
              ? "active-text-box"
              : "text-box"
          }`}
        >
          Engineer
        </Link>
      </div>
      <div className="skill-box">
        <Link
          to={`${url}/technology`}
          className={`${
            pathname.match(/\/about\/skill\/technology/)
              ? "active-text-box"
              : "text-box"
          }`}
        >
          Technology
        </Link>
      </div>
    </div>
  );
};

export default withRouter(SkillHeader);
