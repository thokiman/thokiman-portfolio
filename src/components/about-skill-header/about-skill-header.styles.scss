.header-skill-container {
    margin: auto;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    height: 10%;
    width: 80%;
    position: relative;
    top: 5px;


    .skill-box {
        margin: auto;


        &:hover {
            padding: 4px 0;
            border-bottom: #3f4249 solid 2px;
            animation: skill-box 1s ease-in-out;
        }

        .active-text-box {
            text-decoration: none;
            padding: 4px 0;
            color: #74c6d4;
            font-family: futura-demi;
            letter-spacing: 0.03em;
            border-bottom: #3f4249 solid 2px;

        }

        .text-box {
            padding: 4px 0;
            text-decoration: none;
            color: #74c6d4;
            font-family: futura-demi;
            letter-spacing: 0.03em;

            &:hover {
                color: #e3e1e4;
                opacity: 0.8;
            }
        }
    }
}

@keyframes skill-box {
    from {
        opacity: 0;
        color: #74c6d4;

    }

    to {
        opacity: 0.8;
        color: #e3e1e4;
    }
}