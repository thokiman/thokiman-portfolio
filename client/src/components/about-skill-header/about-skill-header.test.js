import React from "react";
import { shallow } from "enzyme";

import { SkillHeader } from "./about-skill-header.component";

describe("SkillHeader component", () => {
  let wrapper;
  const mockText = "";

  beforeEach(() => {
    const mockProps = {
      location: { pathname: mockText },
      skillRoute: mockText,
      visualArtRoute: mockText,
      engineerRoute: mockText,
      technologyRoute: mockText,
    };
    wrapper = shallow(<SkillHeader {...mockProps} />);
  });

  it("should render SkillHeader component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
