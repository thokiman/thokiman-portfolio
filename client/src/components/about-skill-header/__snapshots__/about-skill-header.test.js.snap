// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillHeader component should render SkillHeader component 1`] = `ShallowWrapper {}`;
